src/Init/Tactics.lean

```diff
@@ -1148,6 +1148,108 @@ macro "haveI" d:haveDecl : tactic => `(tactic| refine_lift haveI $d:haveDecl; ?_
 /-- `letI` behaves like `let`, but inlines the value instead of producing a `let_fun` term. -/
 macro "letI" d:haveDecl : tactic => `(tactic| refine_lift letI $d:haveDecl; ?_)
 
+/--
+Configuration for the `decide` tactic family.
+-/
+structure DecideConfig where
+  /-- If true (default: false), then use only kernel reduction when reducing the `Decidable` instance.
+  This is more efficient, since the default mode reduces twice (once in the elaborator and again in the kernel),
+  however kernel reduction ignores transparency settings. The `decide!` tactic is a synonym for `decide +kernel`. -/
```

Comment:
Maybe `decide!` is young enough that we can remove it in favor of the more explicit, but still convenient, `decide +kernel` syntax? Or is `decide!` fundamental enough that it deserves the short form/

Reply:
That makes sense deprecating `decide!`, since `!` is not really evocative of doing kernel reduction. I'll leave that for a future PR.

Reply:
Sure. But still before the next stable release?